### ISSUE TITLE
Request: On file upload dialog, add button to insert local filename

### ISSUE THREAD

**Issue by AndydeCleyre** [telegramdesktop/tdesktop#5231]
Telegram servers can mangle uploaded filenames, and even when they don't, clients can make it hard to see the most informative segments of long filenames. This can be worked around by copying the filename into the caption.

This action could be made easier. In the case of a single-file upload, a button to immediately insert the local filename into the textbox would be sufficient. This would be on the dialog below:

![image](https://user-images.githubusercontent.com/1787385/46566689-84790f80-c8f1-11e8-980c-f8c1895d4aec.png)

But the current problem is magnified in the case of bulk uploads, which breaks the above approach in two ways:

1. a bulk upload offers one "comment" for the entire set of files
2. a bulk upload sends a message with the comment text, rather than using a caption at all

So for the bulk upload dialog, a better solution would be a checkbox item like "Caption each file with its local filename" -- this dialog:

![image](https://user-images.githubusercontent.com/1787385/46566701-c013d980-c8f1-11e8-97f2-82f4832189a3.png)

Since the bulk upload comment input box doesn't affect the captions, that would stay unchanged in appearance and behavior (send a message preceding the uploads).

Returning to the the single-upload approach, then: If we want to be more consistent with the bulk upload interface, we could use a similar checkbox like "Caption this file with its local filename" -- but in this case, since the caption input box actually controls the caption, it would change in the following ways:

- when the new checkbox is checked, the caption input box is disabled, and the immutable text is set to the local filename
- when the new checkbox is unchecked, the input box enables, and the text remains the local filename, ready for editing

**Comment by AndydeCleyre** [comment]
@Aokromes 

That first one is about showing captions in the chat's file view, not about helping the uploader add captions. 

That second one is about showing uploaded filenames, which doesn't help if the upload process already stripped key info from the name.

**Comment by AndydeCleyre** [comment]
Please re-open, robot. Thank you.

**Comment by AndydeCleyre** [comment]
The mechanics of bulk upload captions may have changed since the initial report.

**Comment by nodeus** [comment]
Or automatically add the name of the local file to the caption field.

**Comment by AndydeCleyre** [comment]
Yes, I'd find that acceptable and handy for the single-file case.
